projects/plugins/crm/includes/ZeroBSCRM.IntegrationFuncs.php

```diff
@@ -885,8 +885,8 @@ function zeroBS_integrations_addOrUpdateTask(
 		is_array( $data_array )
 		&& count( $data_array ) > 0
 		&& ! empty( $data_array['title'] )
-		&& ! empty( $data_array['start'] )
-		&& ! empty( $data_array['end'] )
+		&& ! empty( $data_array['from'] )
+		&& ! empty( $data_array['to'] )
```

Comment:
Great find! Due to other integrations, we'll need to support both versions. How does this look?

```suggestion
		&& (
			// old params
			( ! empty( $data_array['to'] ) && ! empty( $data_array['from'] ) )
			||
			// new params
			( ! empty( $data_array['start'] ) && ! empty( $data_array['end'] ) )
		)
```

Reply:
I don't think there is a need to check for both old and new params in this code (unless we do more code changes)

Reason: I can see the function enclosing this code (zeroBS_integrations_addOrUpdateTask) is only called once in the whole codebase from this line: https://github.com/Automattic/jetpack/blob/8832e428eb65643273e2e1c41116d6681ebdae64/projects/plugins/crm/api/create_event.php#L93

Reply:
Yep, the official API docs use `to` and `from`, but `start` and `end` are the "new" way to do it, matching the database. Down the road when we update the API, those will be the correct params.

At the moment, the only reason the `to` and `from` work is because of a legacy DAL translation function.

Given that we've supported the new way for quite some time, and that third-party integrations may already use this function, removing it could break existing code. Technically the function is not documented, so we can do what we want, but we generally try to not make breaking changes.

Reply:
Thanks for the extra details.

And I did not mean to remove the code / function :) , I was just refering to the only usage for it.

Based on the where things are heading and the roadmap, I think we go with supporting both "old" and "new" params. 

I tested the shared code above https://github.com/Automattic/jetpack/pull/33712#discussion_r1367507504 and it seems to be working as expected 👍

Reply:
> And I did not mean to remove the code / function

No problem! Really we're the ones that broke it (me specifically), so I'm grateful you spotted it and took the initiative to get it working!